### DISCUSSION TITLE
Ability to change metrics URL in docker container

### DISCUSSION THREAD

**Discussion by gr8routdoors** [Apicurio/apicurio-registry#3645] [Q&A] [answered]
### Feature or Problem Description

My organization uses a different URL than the default `/metrics` specified in the Micrometer spec.  I see that the `quarkus-micrometer-registry-prometheus` extension supports the `quarkus.micrometer.export.prometheus.path` property for changing the URL in its [config reference](https://quarkus.io/guides/telemetry-micrometer#configuration-reference), but I've tried settings this directly as an environment variable and a property in a `JAVA_OPTIONS` environment variable for the docker container with no luck.

### Proposed Solution

I'm probably missing something obvious (maybe this exists already), but ideally, all quarkus properties are overrideable by environment variables in the docker container - as this is the most ubiquitous way to configure containers of any type.

### Additional Context

It looks like another usage of `JAVA_OPTIONS` was discussed in this [previous issue](https://github.com/Apicurio/apicurio-registry/issues/829).  However, I couldn't understand the resolution for the docker image without the k8s operator.  I did try mounting a config volume including an `application.properties` under `/home/jboss` as described in the issue, but it wasn't picked up by the application. (I got `/home/jboss` as the PWD from [this dockerfile](https://catalog.redhat.com/software/containers/ubi8/openjdk-17/618bdbf34ae3739687568813?container-tabs=dockerfile).)

Digging through more issues I didn't see the aforementioned quarkus property listed in [this output](https://github.com/Apicurio/apicurio-registry/blob/fcad909cbae589ffc4c4e8f7e163fc7464fda1e6/out.md) from the [property configuration documentation issue](https://github.com/Apicurio/apicurio-registry/issues/2509), so I began to wonder if this was even configurable.

[a comment by an automated account is not shown]

**Comment by gr8routdoors** [comment]
I just tried extending the docker image to explicitly set `JAVA_OPTS` that includes the property and no luck.  I don't see this getting set explicitly anywhere in the project so I'm a bit stumped.

**Comment by carlesarnal** [comment, marked as answer]
> ### Feature or Problem Description
> My organization uses a different URL than the default `/metrics` specified in the Micrometer spec. I see that the `quarkus-micrometer-registry-prometheus` extension supports the `quarkus.micrometer.export.prometheus.path` property for changing the URL in its [config reference](https://quarkus.io/guides/telemetry-micrometer#configuration-reference), but I've tried settings this directly as an environment variable and a property in a `JAVA_OPTIONS` environment variable for the docker container with no luck.
> 
> ### Proposed Solution
> I'm probably missing something obvious (maybe this exists already), but ideally, all quarkus properties are overrideable by environment variables in the docker container - as this is the most ubiquitous way to configure containers of any type.
> 
> ### Additional Context
> It looks like another usage of `JAVA_OPTIONS` was discussed in this [previous issue](https://github.com/Apicurio/apicurio-registry/issues/829). However, I couldn't understand the resolution for the docker image without the k8s operator. I did try mounting a config volume including an `application.properties` under `/home/jboss` as described in the issue, but it wasn't picked up by the application. (I got `/home/jboss` as the PWD from [this dockerfile](https://catalog.redhat.com/software/containers/ubi8/openjdk-17/618bdbf34ae3739687568813?container-tabs=dockerfile).)
> 
> Digging through more issues I didn't see the aforementioned quarkus property listed in [this output](https://github.com/Apicurio/apicurio-registry/blob/fcad909cbae589ffc4c4e8f7e163fc7464fda1e6/out.md) from the [property configuration documentation issue](https://github.com/Apicurio/apicurio-registry/issues/2509), so I began to wonder if this was even configurable.



> I just tried extending the docker image to explicitly set `JAVA_OPTS` that includes the property and no luck. I don't see this getting set explicitly anywhere in the project so I'm a bit stumped.

None of the approaches you tried is going to work for this property in particular. The issue is basically that the property you're trying to modify is a build time property and cannot be modified at runtime. To be able to change it you would need your own build of the application. If you take a look at the [documentation](https://quarkus.io/guides/telemetry-micrometer#quarkus-micrometer_quarkus.micrometer.export.prometheus.path) you can see that there is a lock close to the property.

**Comment by gr8routdoors** [comment]
Got it.  Thanks for the response.
